src/s3ql/fs.py

```diff
@@ -933,8 +947,21 @@ async def open(self, id_, flags, ctx):
             raise FUSEError(errno.EPERM)
 
         if flags & os.O_TRUNC:
-            log.warning('open() with O_TRUNC is not yet supported.')
-            raise FUSEError(errno.ENOTSUP)
+            if flags & os.O_RDWR or flags & os.O_WRONLY:
+                attr = await self.getattr(id_, ctx)
+                if stat.S_ISREG(attr.st_mode):
+                    attr.st_mtime_ns = time_ns()
+                    attr.st_size = 0
+                    await self.setattr(id_, attr, _TruncSetattrFields, None, ctx)
+                elif stat.S_ISFIFO(attr.st_mode) or stat.S_ISBLK(attr.st_mode):
+                    #  silently ignore O_TRUNC when FIFO or terminal block device
+                    pass
+                else:
+                    #  behaviour is not defined in POSIX, we opt for an error
+                    raise FUSEError(errno.EINVAL)
+            else:
```

Comment:
Small style issue: if you handle this case first, you can save an indentation level which makes reading easier.

```
if not (flags & os.O_RDWR) ....
   raise FUSEError(...)
attr = await ...
...
```